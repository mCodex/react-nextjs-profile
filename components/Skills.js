import React, {PureComponent, Fragment} from 'react'
import {Label, Grid} from 'semantic-ui-react'
import {sample} from 'lodash'

const skills = ['NodeJS', 'React', 'React-Native', 'ExpressJS', 'Webpack', 'Serverless Framework', 'Mocha', 'MySQL', 'MongoDB', 'GraphQL', 'Python', 'GNU/Linux', 'GIT' ]

const colors = [
  'red',
  'orange',
  'yellow',
  'olive',
  'green',
  'teal',
  'blue'
]
export default class Skills extends PureComponent {
  render() {
    return(
      <Fragment>
        <Grid stackable columns={13}>
          {skills.map(skill =>
            <Grid.Column>
              <Label tag color={sample(colors)}>{skill}</Label>
            </Grid.Column>
          )}
        </Grid>
      </Fragment>
    )
  }
}
