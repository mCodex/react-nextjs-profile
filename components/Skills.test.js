import React from 'react';
import renderer from 'react-test-renderer';

import Skills from './Skills';

describe('Skills component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Skills />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
