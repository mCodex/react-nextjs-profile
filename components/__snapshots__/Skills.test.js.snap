// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills component renders correctly 1`] = `
<div
  className="ui stackable grid"
>
  <div
    className="ui list"
    role="list"
  >
    <h2
      className="header"
    >
      <span
        className={undefined}
        style={undefined}
      >
        skills
      </span>
    </h2>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      NodeJS
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      React
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      React-Native
    </div>
  </div>
</div>
`;
